Adding texts in verify_leader metric (#11997)
hashicorp/consul: website/content/docs/agent/telemetry.mdx

| `consul.raft.transition.heartbeat_timeout`         | The number of times an agent has transitioned to the Candidate state, after receive no heartbeat messages from the last known leader.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   | timeouts / interval               | counter |

> | `consul.raft.verify_leader`                         | This metric doesn't have direct correlation to the leader change.  It just counts the number of times an agent checks whether it is still the leader or not.  For example, during every consistent read, the check is done.  So, depending on the load in the system, this metric can be high as it is incremented each time a consistent read is done.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | checks / interval                 | Counter |

Reviewer: ```suggestion
| `consul.raft.verify_leader`                         | This metric doesn't have a direct correlation to the leader change.  It just counts the number of times an agent checks if it is still the leader or not.  For example, during every consistent read, the check is done.  Depending on the load in the system, this metric count can be high as it is incremented each time a consistent read is completed.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | checks / interval                 | Counter |
```
Minor language cleanup
